### PR TITLE
plugin/kubernetes: Validate transfers are allowed

### PR DESCRIPTION
**Pull request by chrisohaver**
### 1. Why is this pull request needed and what does it do?

Actually check that the requesting IP is allowed to transfer the zone, as is described in the docs.

### 2. Which issues (if any) are related?
#2286 

### 3. Which documentation changes (if any) need to be made?
none

### CODE REVIEW

File: plugin/kubernetes/xfr.go [hunk 1]
```diff
@@ -71,6 +75,25 @@ func (k *Kubernetes) Transfer(ctx context.Context, state request.Request) (int,
 	return dns.RcodeSuccess, nil
 }
 
+// transferAllowed checks if incoming request for transferring the zone is allowed according to the ACLs.
+func (k *Kubernetes) transferAllowed(state request.Request) bool {
```
Comment (miekg) [inline plugin/kubernetes/xfr.go:80]: Please add a comment that this is copy from zone.transferAllowed and should be (at somepoint) factored into a transfer pkg (or where ever we put it)

File: plugin/kubernetes/xfr_test.go [hunk 2]
```diff
@@ -15,7 +15,7 @@ import (
 func TestKubernetesXFR(t *testing.T) {
 	k := New([]string{"cluster.local."})
 	k.APIConn = &APIConnServeTest{}
-	k.TransferTo = []string{"127.0.0.1"}
```
Comment (miekg) [inline plugin/kubernetes/xfr_test.go:18]: if you don't specify the port in the setup, it still works?

Comment (chrisohaver) [reply plugin/kubernetes/xfr_test.go:18]: I recall that it runs it through a normalizer, which adds the default 53 port...

I'll double check  ...

**Comment by chrisohaver** [comment]
Added tests for transfer config parsing (there were none).  Verifies address with/without port.

File: plugin/kubernetes/xfr_test.go (hunk 2 above)
Comment (bradbeam) [reply plugin/kubernetes/xfr_test.go:18]: https://github.com/coredns/coredns/blob/master/plugin/pkg/parse/host.go#L82 Yeah it should be fine to specify w/o a port.

**Review by bradbeam — APPROVED** [review]

**Pull request merged**